node/src/components/storage/tests.rs

```diff
@@ -2071,3 +2074,55 @@ fn should_get_block_header_by_height() {
     assert!(maybe_block_header.is_some());
     assert_eq!(expected_header, maybe_block_header.unwrap());
 }
+
+#[test]
+fn should_read_legacy_unbonding_purse() {
+    // These bytes represent the `UnbondingPurse` struct with the `new_validator` field removed
+    // and serialized with `bincode`.
+    // In theory, we can generate these bytes by serializing the `WithdrawPurse`, but at some point,
+    // these two structs may diverge and it's a safe bet to rely on the bytes
+    // that are consistent with what we keep in the current storage.
+    const LEGACY_BYTES: &str = "0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e0e07010000002000000000000000197f6b23e16c8532c6abc838facd5ea789be0c76b2920334039bfa8b3d368d610100000020000000000000004508a07aa941707f3eb2db94c8897a80b2c1197476b6de213ac273df7d86c4ffffffffffffffffff40feffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+
+    let decoded = hex::decode(LEGACY_BYTES).expect("decode");
```

Comment:
We normally use the `base16` crate for this - that will avoid you importing `uint::hex`.

Reply:
Fixed in 0c32238bf66d3d14721e45c9c88d31073af8e9c3